src/controllers/badgeController.js

```diff
@@ -1,11 +1,11 @@
 const mongoose = require('mongoose');
 const UserProfile = require('../models/userProfile');
-const { hasPermission } = require('../utilities/permissions');
+const { hasPermission, hasIndividualPermission } = require('../utilities/permissions');
 const escapeRegex = require('../utilities/escapeRegex');
 
 const badgeController = function (Badge) {
   const getAllBadges = async function (req, res) {
-    if (!await hasPermission(req.body.requestor.role, 'seeBadges')) {
+    if (!await hasPermission(req.body.requestor.role, 'seeBadges') && !await hasIndividualPermission(req.body.requestor.requestorId, 'seeBadgeManagementTab')) {
```

Comment:
```suggestion
    if (!await hasPermission(req.body.requestor, 'seeBadges') && !await hasPermission(req.body.requestor, 'seeBadgeManagementTab')) {
```
Probably need to pull the latest changes. The arguments have changed.